hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientReliableTopicProxy.java

```diff
@@ -114,10 +114,9 @@ public void publish(E payload) {
                 default:
                     throw new IllegalArgumentException("Unknown overloadPolicy:" + overloadPolicy);
             }
-        } catch (RuntimeException e) {
-            throw e;
         } catch (Exception e) {
```

Comment:
The code in `try` block above may throw a `RuntimeException` or an `ExecutionException` wrapping the cause, so `peel`ing it will rethrow one of:
- the `RuntimeException` itself
- the cause of `ExecutionException`, when that is an instance of `RuntimeException`
- a `HazelcastException` wrapping a non-`RuntimeException`

With the previous `catch` block, any non-`RuntimeException` would be thrown wrapped in a `HazelcastException`, resulting in the `HazelcastException`->`ExecutionException`->actual exception wrapping observed in #9960